ERP-BACKEND/app/services/search_service.py

```
@@ -21,40 +21,120 @@ def __init__(self, db: Session, use_elasticsearch: bool = False):
+                    "searchable_text": item.get("searchable_text", f"{item['title']} {item['content']}"),
+                    "meta_data": item.get("meta_data", {}),
+                    "tags": item.get("tags", []),
+                    "created_at": datetime.utcnow(),
+                    "updated_at": datetime.utcnow()
+                })
+            
+            # Use bulk insert with upsert on conflict
+            stmt = postgresql_insert(SearchIndex).values(values_list).on_conflict_do_update(
+                index_elements=['entity_type', 'entity_id'],
+                set_={
+                    'title': stmt.excluded.title,
```

Ruff (0.16.1)

[error] Undefined name `stmt`

(F821)